### DISCUSSION TITLE
Deleting project/scenario/item

### DISCUSSION THREAD

**Discussion by fturibak** [ludeknovy/jtl-reporter#99] [General]
Hello!

I've figured out that if you have crete custom chart and set some metrics, it's impossible to delete that item because of constraints in database.  

**ERROR:  update or delete on table "items" violates foreign key constraint "user_item_chart_settings_item_id_fkey" on table "user_item_chart_settings"**

Consequently you can't delete scenario/project of belonging item.

UI reports only general error and it's pretty hard to trace error.

One way is to "delete" custom chart via UI or via database query, but if you have hundred of items in hundred of scenarios it's pain in the ass. Beg my pardon  :-)

It would be nice if **DELETE** method for items/scenarios/projects would clean-up all records in table **user_item_chart_settings** belonging to item that's being deleted.

**Comment by ludeknovy** [comment]
Hello @fturibak. 

I created an issue from this discussion. Agree with you, it must be a nightmare to delete dozens of items in such a case ;)

Thanks for reporting.

**Reply by ludeknovy** [reply]
It will be fixed in v4.
